fastlane/lib/fastlane/actions/jira.rb

```diff
@@ -21,18 +25,34 @@ def self.run(params)
                     password: password
                   }
 
-        client = JIRA::Client.new(options)
-        issue = client.Issue.find(ticket_id)
-        comment = issue.comments.build
-        comment.save({ 'body' => comment_text })
+        begin
+          client = JIRA::Client.new(options)
+          issue = client.Issue.find(ticket_id)
+          comment = issue.comments.build
+          comment.save({ 'body' => comment_text })
+
+          # An exact representation of the JSON returned from the JIRA API
+          # https://github.com/sumoheavy/jira-ruby/blob/master/lib/jira/base.rb#L67
+          json_response = comment.attrs
+          if json_response.nil?
+            UI.error('Failed to add a comment on JIRA ticket')
+            return nil
+          else
+            Actions.lane_context[SharedValues::JIRA_JSON] = json_response
+            UI.success('Successfully added a comment on JIRA ticket')
+            return json_response
+          end
+        rescue => exception
+          UI.error("Exception: #{exception}")
+        end
```

Comment:
I don’t think we’d want to wrap this in a begin/rescue here 🤔 This is a breaking change for anybody that is actually expecting this to fail and halt the _fastlane_ process.

You _could_ instead update your `Fastfile` to look like...

```ruby
lane :do_thing do
  begin
    jira
  rescue
    UI.error(“Failed to add a comment on JIRA ticket”)
  end
end
```

I think this _might_ be a better approach as its more flexible and non-breaking 😇 I would like to hear your thoughts on this though!

Reply:
Thank you very much for your time to review the PR and provide me this valuable suggestion. 🙇‍♂️
I agree we should not break the existing flow. Added one new option `fail_on_error` with **default value** true

- [x] Now, for existing users, On error, it will halt the fastlane process. 'Backward compatibility'
- [x] Also, now users can use **FL_JIRA_FAIL_ON_ERROR** in the DotEnv file to control the flow. 😇

Requesting you for re-review. Thank you very much again! 🙇‍♂️